src/fides/api/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -478,3 +487,50 @@ def send_test_message(
             status_code=400, detail=f"There was an error sending the test message: {e}"
         )
     return {"details": "Test message successfully sent"}
+
+
+@router.get(
+    MESSAGING_TEMPLATES,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=List[MessagingTemplateResponse],
+)
+def get_messaging_templates(
+    *, db: Session = Depends(deps.get_db)
+) -> List[MessagingTemplateResponse]:
+    """Returns the available messaging templates, augments the models with labels to be used in the UI."""
+    return [
+        MessagingTemplateResponse(
+            key=template.key,
+            content=template.content,
+            label=DEFAULT_MESSAGING_TEMPLATES.get(template.key, {}).get("label", None),
+        )
+        for template in get_all_messaging_templates(db=db)
+    ]
+
+
+@router.post(
+    MESSAGING_TEMPLATES,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+)
+def update_messaging_templates(
+    templates: List[MessagingTemplateRequest], *, db: Session = Depends(deps.get_db)
+) -> Dict[str, Any]:
+    """Updates the messaging templates and reverts empty subject or body values to the default values."""
+    for template in templates:
+        key = template.key
+        content = template.content
+
+        default_template = DEFAULT_MESSAGING_TEMPLATES.get(key)
+        if default_template:
+            content["subject"] = (
+                content.get("subject") or default_template["content"]["subject"]
+            )
+            content["body"] = content.get("body") or default_template["content"]["body"]
+
+            MessagingTemplate.create_or_update(
+                db, data={"key": key, "content": content}
+            )
+        else:
+            logger.debug("Invalid template key: {}, skipping creation.", key)
```

Comment:
Should this also raise an error?

If I tried saving a non default messaging template the response of the endpoint would be `{"message": "Email templates updated successfully."}` despite it not actually saving

Reply:
I did this so if you had one bad template but the other ones were good, the entire request didn't fail.

Reply:
That makes sense. This might be too much for this endpoint but we do have some patch endpoints that keep track of what records successfully updated and what ones failed.


If the chances of someone submitting an invalid template is extremely low then this is probably okay for now considering it's mainly the UI that's using it. That being said I still think it's a surprising behavior to send back a success message when not everything succeeded.

Reply:
Updated this endpoint to return a `BulkPutMessagingTemplateResponse`.